### DISCUSSION TITLE
Flutter "create" type cast exception

### DISCUSSION THREAD

**Discussion by vojtarambo** [pocketbase/pocketbase#4532] [Q&A]
I am having some trouble with the "create" method in PocketBase collection. When calling, I get "type 'String' is not a subtype of type 'Map<String, dynamic>?' in type cast" exception. This happens even when I call "create" without the body parameter or when I set the parameter to "{}". This means that my function that converts the data to Map is not the problem. The strange thing is that this occurs only when I test on an actual Android device. When I test on an emulator, it works fine. The problem that Flutter dumps is in the file "pocketbase/src/services/base_crud_service.dart:186:46". Other functions (getFullList, getOne,...) work well as well as create when I send a request manually.

Flutter version: 3.19.3
Dart vesion: 3.3.1
PocketBase vesion: 0.18.1
Android version: 14

My function call
![call](https://github.com/pocketbase/pocketbase/assets/74298402/c6a8cdab-c528-4582-8366-f6dbfadfa6ef)

The error I get
![error](https://github.com/pocketbase/pocketbase/assets/74298402/297b6293-5d0b-4d0c-a805-7b7a297798b5)

The line that causes the problem according to Flutter
![problem](https://github.com/pocketbase/pocketbase/assets/74298402/09b8e460-36e3-4de1-8f09-8ed5d900ae0e)

**Comment by ganigeorgiev** [comment]
> This means that my function that converts the data to Map is not the problem. 

I have no way to verify this. Please provide a **minimal** reproducible repo or at least steps to reproduce with a more detailed code sample.

**Reply by ganigeorgiev** [reply]
Oh, actually your error may not be caused by the data you send as body but the response.

Make sure that your Android application can connect to the PocketBase server (aka. it is not resulting in timeout or some other similar non json error response; or you are also may be using incorrect base url that results in 200 plain text response).

**Reply by ganigeorgiev** [reply]
To verify that your Android app can reach PocketBase, you can start the executable with `--dev` and check the logs in the console and/or inspect the logs in the _Admin UI > Logs_. If there is no request logged in, then most likely your app cannot reach the PocketBase server (**in that case make sure that you use the correct base url when initializing the SDK**; if you are testing in a emulator or some other vm environment not that the mobile device localhost is different from the host machine; see also https://developer.android.com/studio/run/emulator-networking).

**Comment by vojtarambo** [comment]
Hey, so I managed to resolve this by switching to a different server. The first one was working fine when viewing or listing records in the database, the error occurred only when trying to create a record. On the new hosting is everything working well. I have no idea what exactly caused the error, but I guess that there is some edge-case bug in the Flutter implementation.

**Reply by ganigeorgiev** [reply]
Please check the above comment in https://github.com/pocketbase/pocketbase/discussions/4532#discussioncomment-8722294 to help you with the debugging.

> but I guess that there is some edge-case bug in the Flutter implementation.

I'm not ruling out a PocketBase issue but without a minimal reproducible repo or at least steps to reproduce I'm not sure how to help you. If you are still experiencing the issue feel free to provide more details by opening an issue/PR in the [Dart SDK repo](https://github.com/pocketbase/dart-sdk) and I'll try to investigate it further.
